lib/src/sql/datetime.rs

```diff
@@ -34,11 +36,39 @@ impl From<DateTime<Utc>> for Datetime {
 	}
 }
 
-impl From<&str> for Datetime {
-	fn from(s: &str) -> Self {
-		match datetime_all_raw(s) {
-			Ok((_, v)) => v,
-			Err(_) => Self::default(),
+impl From<Datetime> for DateTime<Utc> {
+	fn from(x: Datetime) -> Self {
+		x.0
+	}
+}
+
+impl FromStr for Datetime {
+	type Err = ();
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		Self::try_from(s)
+	}
+}
+
+impl TryFrom<String> for Datetime {
+	type Error = ();
+	fn try_from(v: String) -> Result<Self, Self::Error> {
+		Self::try_from(v.as_str())
+	}
+}
+
+impl TryFrom<Strand> for Datetime {
+	type Error = ();
+	fn try_from(v: Strand) -> Result<Self, Self::Error> {
+		Self::try_from(v.as_str())
+	}
+}
+
+impl TryFrom<&str> for Datetime {
+	type Error = ();
+	fn try_from(v: &str) -> Result<Self, Self::Error> {
+		match datetime_all_raw(v) {
+			Ok((_, v)) => Ok(v),
+			_ => Err(()),
 		}
 	}
 }
```

Comment:
These seem a bit redundant. While I don't think they should delay this PR, I might try to consolidate them in a future PR.